src/test/validation_block_tests.cpp

```diff
@@ -204,14 +204,12 @@ BOOST_AUTO_TEST_CASE(processnewblock_signals_ordering)
     for (auto& t : threads) {
         t.join();
     }
-    while (GetMainSignals().CallbacksPending() > 0) {
-        UninterruptibleSleep(std::chrono::milliseconds{100});
-    }
+    SyncWithValidationInterfaceQueue();
```

Comment:
This is part of the bugfix or an optimization? Or a different bugfix?

Reply:
Thanks, split the optimization into a separate commit